Make sure error messages actually make it out

These error messages appeared in the logger text:

```
2025-01-30 07:06:55,127 - servicex_local.science_images - DEBUG - [ 50%] Building CXX object analysis/CMakeFiles/analysisLib.dir/Root/query.cxx.o
2025-01-30 07:06:58,380 - servicex_local.science_images - DEBUG - /home/atlas/rel/source/analysis/Root/query.cxx: In member function 'virtual StatusCode query::execute()':
2025-01-30 07:06:58,381 - servicex_local.science_images - ERROR - /home/atlas/rel/source/analysis/Root/query.cxx:93:48: error: 'Muon_v1' is not a member of 'xAOD'
2025-01-30 07:06:58,381 - servicex_local.science_images - DEBUG - 93 |       std::vector<ElementLink<DataVector<xAOD::Muon_v1>>> auxdataConst_vector_ElementLink_DataVector_xAOD_Muon_v1___3;
2025-01-30 07:06:58,381 - servicex_local.science_images - DEBUG - |                                                ^~~~~~~
```

But nothing was dumped to the jupyter notebook - so some how the logging messages are getting swallowed.

--- a/servicex_local/logging_decorator.py
+++ b/servicex_local/logging_decorator.py
@@ -7,7 +7,9 @@ def log_to_file(log_file):
         @wraps(func)
         def wrapper(*args, **kwargs):
             logger = logging.getLogger()
+            original_logging_level = logger.level
             logger.setLevel(logging.DEBUG)
+
             handler = logging.FileHandler(log_file, mode="a")  # Append mode
             formatter = logging.Formatter(
                 "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
@@ -19,6 +21,7 @@ def log_to_file(log_file):
             finally:
                 logger.removeHandler(handler)
                 handler.close()
+                logger.setLevel(original_logging_level)
 
         return wrapper
 

--- a/servicex_local/science_images.py
+++ b/servicex_local/science_images.py
@@ -3,7 +3,7 @@ import os
 import subprocess
 from abc import ABC, abstractmethod
 from pathlib import Path
-from typing import List
+from typing import List, Optional
 
 from .logging_decorator import log_to_file
 
@@ -35,15 +35,27 @@ def run_command_with_logging(command: List[str], log_file: Path) -> None:
 
         assert process.stdout is not None
 
+        emit_next_line_level: Optional[int] = None
         for stdout_line in iter(process.stdout.readline, ""):
             stripped_line = stdout_line.strip()
             stdout_lines.append(stripped_line)
-            if "error" in stripped_line.lower():
+            emitted_level: Optional[int] = None
+            if (emit_next_line_level == logging.ERROR) or (
+                "error" in stripped_line.lower()
+            ):
                 logger.error(stripped_line)
-            elif "warning" in stripped_line.lower():
+                emitted_level = logging.ERROR
+            elif (emit_next_line_level == logging.WARNING) or (
+                "warning" in stripped_line.lower()
+            ):
                 logger.warning(stripped_line)
+                emitted_level = logging.WARNING
             else:
                 logger.debug(stripped_line)
+
+            emit_next_line_level = None
+            if stripped_line.endswith(":"):
+                emit_next_line_level = emitted_level
 
         process.stdout.close()
         return_code = process.wait()

--- a/tests/test_logging_decorator.py
+++ b/tests/test_logging_decorator.py
@@ -24,3 +24,22 @@ def test_logfile_written(tmp_path: Path, caplog: LogCaptureFixture):
 
     lines = [ln for ln in log.read_text().split("\n") if len(ln.strip()) != 0]
     assert len(lines) == 4
+
+
+def test_logfile_no_mess_with_levels(tmp_path: Path, caplog: LogCaptureFixture):
+    log = tmp_path / "func_log.txt"
+
+    @log_to_file(log)
+    def run_me():
+        logging.info("info")
+        logging.debug("debug")
+        logging.warning("warning")
+        logging.error("error")
+
+    root_logger = logging.getLogger()
+    with caplog.at_level(logging.WARNING):
+        assert root_logger.level == logging.WARNING
+        run_me()
+        assert root_logger.level == logging.WARNING
+
+    assert root_logger.level == logging.WARNING

--- a/tests/test_logging_decorator.py
+++ b/tests/test_logging_decorator.py
@@ -1,0 +1,26 @@
+import logging
+from pathlib import Path
+
+from pytest import LogCaptureFixture
+from servicex_local.logging_decorator import log_to_file
+
+
+def test_logfile_written(tmp_path: Path, caplog: LogCaptureFixture):
+    log = tmp_path / "func_log.txt"
+
+    @log_to_file(log)
+    def run_me():
+        logging.info("info")
+        logging.debug("debug")
+        logging.warning("warning")
+        logging.error("error")
+
+    with caplog.at_level(logging.WARNING):
+        run_me()
+
+    assert len(caplog.records) == 2
+    for r in caplog.records:
+        assert r.msg in ["warning", "error"]
+
+    lines = [ln for ln in log.read_text().split("\n") if len(ln.strip()) != 0]
+    assert len(lines) == 4

